core/src/execution/evm.rs

```diff
@@ -313,18 +317,18 @@ impl<N: NetworkSpec> Database for ProofDB<N> {
         Ok(Some(self.state.get_basic(address)?))
     }
 
-    fn block_hash(&mut self, number: u64) -> Result<B256, Report> {
+    fn block_hash(&mut self, number: u64) -> Result<B256, DatabaseError> {
         trace!(target: "helios::evm", "fetch block hash for block={:?}", number);
         self.state.get_block_hash(number)
     }
 
-    fn storage(&mut self, address: Address, slot: U256) -> Result<U256, Report> {
+    fn storage(&mut self, address: Address, slot: U256) -> Result<U256, DatabaseError> {
         trace!(target: "helios::evm", "fetch evm state for address={:?}, slot={}", address, slot);
         self.state.get_storage(address, slot)
     }
 
-    fn code_by_hash(&mut self, _code_hash: B256) -> Result<Bytecode, Report> {
-        Err(eyre::eyre!("should never be called"))
+    fn code_by_hash(&mut self, _code_hash: B256) -> Result<Bytecode, DatabaseError> {
+        unimplemented!("should never be called")
```

Comment:
Just to be extra safe here lets return an error so that if revm's behavior changes Helios doesn't crash, but instead errors. We shouldn't use the StateMissing error though, just to avoid an infinite loop of trying to refetch the state if this triggers.